app/scripts/stores/dashboardStore.js

```diff
@@ -0,0 +1,290 @@
+        drawerPosition ? contentStyle.marginLeft = this.drawer.get('width') : null
+        this.drawer.set('open', drawerPosition)
+        this.drawer.set('contentStyle', contentStyle)
+    }
+    
+    @action setDrawer() {
+        let width = 350;
+        let contentStyle = { transition: 'margin-left 450ms cubic-bezier(0.23, 1, 0.32, 1)' };
+        contentStyle.marginLeft = width;
+        this.drawer.set('open', true);
+        this.drawer.set('width', width);
+        this.drawer.set('contentStyle', contentStyle);
+    }
+    
+    @action toggleCollapseTree(router) {
+        if (this.drawer.get('collapsed')) {
+            this.downloadedItems.forEach((item) => {
+                item.open = true
+            })
+            this.drawer.set('collapsed', false);
+            this.drawer.set('toggleLable', 'Home');
+        } else {
+            mainStore.listItems = mainStore.projects
+            this.selectedItem = null
+            this.downloadedItems.forEach((item) => {
+                item.open = false
+            })
+            router.push({pathname: ("/dashboard")})
+            this.drawer.set('collapsed', true);
+            this.drawer.set('toggleLable', 'Expand');
+        }
+    }
+    
+    @action toggleTreeListItem(id) {
+        this.selectedItem = id;
+        let item = this.downloadedItems.get(id)
+        item.open = !item.open
+        this.downloadedItems.delete(id)
+        this.downloadedItems.set(id, item)
+    }
+
+    @action getItem(id, path, isSelected) {
+        mainStore.loading = true;
+        this.transportLayer.getEntity(id, path)
+            .then(checkStatusAndConsistency)
+            .then(response => response.json())
+            .then((json) => {
+                if(!json.error) {
+                    let item = this.downloadedItems.get(id)
+                    if(item) {
+                        json.open = true
+                        json.childrenIds = item.childrenIds
+                        json.folderIds = item.folderIds
+                        this.downloadedItems.delete(id)
+                    }
+                    this.downloadedItems.set(id, json)
+                    if(isSelected) this.selectedItem = id
+                    mainStore.loading = false;
+                } else {
+                    json.code === 'resource_not_consistent' ? mainStore.tryAsyncAgain(getItem, [id, path, isSelected]) : mainStore.handleErrors(json);
+                }
+            }).catch(ex => mainStore.handleErrors(ex))
+    }
+
+    @action setSelectedItem() {
+        let {id, path} = this.router.params
+        if (id && path) {
+            if (this.downloadedItems.has(id)) {
+                this.selectedItem = id
+            } else {
+                this.getItem(id, `${path}/`, true)
+            }
+        }
+    }
+
+    @action selectItem(itemId) {
+        this.drawer.set('toggleLable', 'Home');
+        mainStore.filesChecked = []
```

Comment:
I actually want to refactor how this is done in the mainStore. Not necessary for this pull request though. I don't like how I used arrays of file or folder id's when items are selected in the listItems.jsx component. Not sure why I did it this way before. Should be using an observable.map() or even just an array of selected items where each item in the array is the file or folder object with all properties. Then filter by kind as needed. I will refactor this after we pull in your changes.